tests/fips/mozilla_nss/firefox_nss.pm

```diff
@@ -9,78 +9,64 @@
-    if (check_screen("firefox-save-and-quit", 10)) {
-        send_key "ret";
-    }
-}
-
 sub run() {
     # define fips password for firefox, and it should be consisted by:
     # - at least 8 characters
     # - at least one upper case
     # - at least one non-alphabet-non-number character (like: @-.=%)
     my $fips_password = 'openqa@SUSE';
-    ensure_installed('mozilla-nss-tools');
-
-    # launch firefox first to generate its default profile
-    x11_start_program("firefox");
-    assert_screen("firefox-launch", 90);    # launch firefox need time
-    quit_firefox;
-
-    # enable fips mode for mozilla firefox
-    x11_start_program("xterm");
-    validate_script_output 'echo | modutil -dbdir ~/.mozilla/firefox/*.default -fips true', sub { m/FIPS mode enabled/ };
-    type_string "killall xterm\n";
 
-    # default fips password is empty and MUST required to input when firefox launch
+    # launch firefox first and enable FIPS mode
     x11_start_program("firefox");
-    assert_screen("firefox-fips-password-inputfiled", 90);
-    send_key "ret";
-    assert_screen("firefox-homepage", 60);
-
-    # change default password to defined one
+    assert_screen "firefox-launch", 90;
     send_key "alt-d";
     type_string "about:preferences#security\n";
     assert_screen "firefox-preferences-security";
-    send_key "alt-shift-m";
+    send_key "alt-shift-u";
     assert_screen "firefox-passwd-master_setting";
-    send_key "tab";
     type_string $fips_password;
     send_key "tab";
     type_string $fips_password;
     send_key "ret";
     assert_screen "firefox-password-change-succeeded";
     send_key "ret";
-    quit_firefox;
+    send_key "alt-d";
+    type_string "about:preferences#advanced\n";
+    assert_and_click "firefox-ssl-advanced_certificate";
+    wait_still_screen;
+    send_key "alt-shift-d";
+    assert_screen "firefox-device-manager";
+    send_key "alt-f";
+    assert_screen "firefox-fips-password-inputfiled", 90;
+    type_string $fips_password;
+    send_key "ret";
+    send_key "tab";
+    send_key "ret";
+    send_key "ctrl-q";
+    wait_still_screen;
 
-    # launch firefox with new password
+    # launch firefox again and check FIPS mode is enabled
     x11_start_program("firefox");
-    assert_screen("firefox-fips-password-inputfiled", 90);
+    assert_screen "firefox-fips-password-inputfiled", 90;
     type_string $fips_password;
     send_key "ret";
-    assert_screen("firefox-homepage", 60);
-
-    # confirm fips has been enabled in firefox
+    assert_screen "firefox-homepage", 90;
     send_key "alt-d";
     type_string "about:preferences#advanced\n";
     assert_screen "firefox-preferences-advanced";
-    assert_and_click "firefox-ssl-advanced_certificate";
-    wait_still_screen(5);
     send_key "alt-shift-d";
     assert_screen "firefox-device-manager";
     send_key "down";
     assert_screen "firefox-confirm-fips_enabled";
     send_key "ret";
-    quit_firefox;
+    send_key "ctrl-q";
```

Comment:
Meant to reduce the use of needle and simplify the test code. The afterward "wait_still_screen;" will add some time buffer for firefox closing. It couldn't make sure the firefox is closed in time, but the old way can't promise the firefox is properly turned off neither.
Should I replace implicit "wait_still_screen;" with assert_screen "generic-desktop" ?